app/main.py

```diff
@@ -1,3 +1,21 @@
+import functools
+
+
 def cache(func):
-    # Write your code here
+
+    memory = {}
+
+    @functools.wraps(func)
+    def wrapper(*args):
+        nonlocal memory
+        if args in memory and memory[args][0] == func.__name__:
```

Comment:
You don't actually need to remember or check function name, because decorator creates a new dictionary for each function (line 6). 

It would be needed if memory was a global variable. In that case dictionary would be the same for all decorated functions.